deepchem/feat/molecule_featurizers/mol_graph_conv_featurizer.py

```diff
@@ -0,0 +1,196 @@
+  List[float]
+    A one-hot vector of the bond feature.
+  """
+  bond_type = get_bond_type_one_hot(bond)
+  same_ring = get_bond_is_in_same_ring_one_hot(bond)
+  conjugated = get_bond_is_conjugated_one_hot(bond)
+  stereo = get_bond_stereo_one_hot(bond)
+  return bond_type + same_ring + conjugated + stereo
+
+
+class MolGraphConvFeaturizer(MolecularFeaturizer):
+  """This class is a featurizer of gerneral graph convolution networks for molecules.
+
+  The default node(atom) and edge(bond) representations are based on
+  `WeaveNet paper <https://arxiv.org/abs/1603.00856>`_. If you want to use your own representations,
+  you could use this class as a guide to define your original Featurizer. In many cases, it's enough
+  to modify return values of `construct_atom_feature` or `construct_bond_feature`.
+
+  The default node representation are constructed by concatenating the following values,
+  and the feature length is 38.
+
+  - Atom type: A one-hot vector of this atom, "C", "N", "O", "F", "P", "S", "Br", "I", "other atoms".
+  - Chirality: A one-hot vector of the chirality, "R" or "S".
+  - Formal charge: Integer electronic charge.
+  - Partial charge: Calculated partial charge.
+  - Ring sizes: A one-hot vector of the number of rings (3-8) that include this atom.
+  - Hybridization: A one-hot vector of "sp", "sp2", "sp3".
+  - Hydrogen bonding: A one-hot vector of whether this atom is a hydrogen bond donor or acceptor.
+  - Aromatic: A one-hot vector of whether the atom belongs to an aromatic ring.
+  - Degree: A one-hot vector of the degree (0-5) of this atom.
+  - Number of Hydrogens: A one-hot vector of the number of hydrogens (0-4) that this atom connected.
+
+  The default edge representation are constructed by concatenating the following values,
+  and the feature length is 11.
+
+  - Bond type: A one-hot vector of the bond type, "single", "double", "triple", or "aromatic".
+  - Same ring: A one-hot vector of whether the atoms in the pair are in the same ring.
+  - Conjugated: A one-hot vector of whether this bond is conjugated or not.
+  - Stereo: A one-hot vector of the stereo configuration of a bond.
+
+  If you want to know more details about features, please check the paper [1]_ and
+  utilities in deepchem.utils.graph_conv_utils.py.
+
+  Examples
+  --------
+  >>> smiles = ["C1CCC1", "C1=CC=CN=C1"]
+  >>> featurizer = MolGraphConvFeaturizer()
+  >>> out = featurizer.featurize(smiles)
+  >>> type(out[0])
+  <class 'deepchem.feat.graph_data.GraphData'>
+  >>> out[0].num_node_features
+  38
+  >>> out[0].num_edge_features
+  11
+
+  References
+  ----------
+  .. [1] Kearnes, Steven, et al. "Molecular graph convolutions: moving beyond fingerprints."
+     Journal of computer-aided molecular design 30.8 (2016):595-608.
+
+  Notes
+  -----
+  This class requires RDKit to be installed.
+  """
+
+  def __init__(self, add_self_loop: bool = False):
+    """
+    Parameters
+    ----------
+    add_self_loop: bool, default False
```

Comment:
Perhaps `add_self_edges` would be clearer?  This isn't really about loops, so the name could be confusing.

Reply:
I agreed! I fixed.